lightningd/invoice.c

```diff
@@ -337,21 +328,48 @@ static void json_listinvoice(struct command *cmd,
 		return;
 	}
 
-
-	json_array_start(response, NULL);
-	json_add_invoices(response, &invs->invlist, buffer, label);
+	if (modern) {
+		json_object_start(response, NULL);
+		json_array_start(response, "invoices");
+	} else
+		json_array_start(response, NULL);
+	json_add_invoices(response, &invs->invlist, buffer, label, modern);
 	json_array_end(response);
+	if (modern)
+		json_object_end(response);
 	command_success(cmd, response);
 }
 
+/* FIXME: Deprecated! */
+static void json_listinvoice(struct command *cmd,
+			     const char *buffer, const jsmntok_t *params)
+{
+	return json_listinvoice_internal(cmd, buffer, params, false);
+}
+
 static const struct json_command listinvoice_command = {
 	"listinvoice",
 	json_listinvoice,
-	"Show invoice {label} (or all, if no {label}))",
-	"Returns an array of {label}, {payment_hash}, {msatoshi} (if set), {complete}, {pay_index} (if paid) and {expiry_time} on success. "
+	"(DEPRECATED) Show invoice {label} (or all, if no {label}))",
+	"Returns an array of {label}, {payment_hash}, {msatoshi} (if set), {complete}, {pay_index} (if paid) and {expiry_time} on success. ",
+	.deprecated = true
 };
 AUTODATA(json_command, &listinvoice_command);
 
+static void json_listinvoices(struct command *cmd,
+			     const char *buffer, const jsmntok_t *params)
+{
+	return json_listinvoice_internal(cmd, buffer, params, true);
+}
+
+static const struct json_command listinvoices_command = {
+	"listinvoices",
+	json_listinvoices,
+	"Show invoice {label} (or all, if no {label}))",
+	"Returns an array of {label}, {payment_hash}, {msatoshi} (if set), {complete}, {pay_index} (if paid) and {expiry_time} on success. ",
```

Comment:
`listinvoices` uses modern so no `compllete`. Also not mentioning `msatoshi_received`.